org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -115,7 +117,10 @@ public void connectClient(JavaLanguageClient client) {
 	@Override
 	public void exit() {
 		logInfo(">> exit");
-		System.exit(0);
+		Executors.newSingleThreadScheduledExecutor().schedule(() -> {
+			logInfo("Forcing exit after 1 min.");
+			System.exit(0);
```

Comment:
wonder if we should return a value other than 0 in this case. It is not actually a clear exit.